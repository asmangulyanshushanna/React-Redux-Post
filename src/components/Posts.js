import React from 'react';
import {useEffect} from 'react';
import {connect} from 'react-redux';
import {fetchPost} from '../actions/postAction'


export function Posts(props) {
  useEffect(() => {
    props.fetchPost()
  }, [])
    return props.posts.map(post => (
      <div key={post.id}>
        <h3>
          {post.title}
        </h3>
        <p>
          {post.body}
        </p>
      </div>
    ))
  }


  const mapStateToProps = (state) => console.log(state, 'STATE')||({
    posts: state.posts.items
  })
  
  
  export default connect(mapStateToProps, {fetchPost})(Posts);
  