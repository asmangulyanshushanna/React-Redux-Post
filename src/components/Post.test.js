import React from 'react';
import { shallow, configure } from 'enzyme';
import { Posts } from './Posts';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()})

describe('posts component', () => {
    
    const fetchPost = jest.fn();
    const props = {
        posts: [{title: 'title', body: 'text'}],
        fetchPost
    }
    const wrapper = shallow(<Posts {...props} />)
    
    it('should render and match a stapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });
    it('should contain title', () => {
        expect(wrapper.find('h3').text()).toEqual('title')
    });
    it('should contain text', () => {
        expect(wrapper.find('p').text()).toEqual('text')
    })
})