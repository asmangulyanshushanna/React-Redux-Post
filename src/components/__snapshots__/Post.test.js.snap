// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts component should render and match a stapshot 1`] = `ShallowWrapper {}`;
